atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -0,0 +1,308 @@
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.plugins.pre_flatpak_create_dockerfile import get_flatpak_source_info
+from atomic_reactor.util import get_exported_image_metadata
+
+
+# Returns flatpak's name for the current arch
+def get_arch():
+    return subprocess.check_output(['flatpak', '--default-arch']).strip()
+
+
+# add_app_prefix('org.gimp', 'gimp, 'gimp.desktop') => org.gimp.desktop
+# add_app_prefix('org.gnome', 'eog, 'eog.desktop') => org.gnome.eog.desktop
+def add_app_prefix(app_id, root, full):
+    prefix = app_id
+    if prefix.endswith('.' + root):
+        prefix = prefix[0:-(1 + len(root))]
+    return prefix + '.' + full
+
+
+def find_desktop_files(builddir):
+    desktopdir = os.path.join(builddir, 'files/share/applications')
+    for (dirpath, dirnames, filenames) in os.walk(desktopdir):
+        for filename in filenames:
+            if not filename.endswith('.desktop'):
+                continue
+            yield os.path.join(dirpath, filename)
+
+
+def find_icons(builddir, name):
+    icondir = os.path.join(builddir, 'files/share/icons/hicolor')
+    for (dirpath, dirnames, filenames) in os.walk(icondir):
+        for filename in filenames:
+            if not filename.startswith(name + '.'):
+                continue
+            yield os.path.join(dirpath, filename)
+
+
+def update_desktop_files(app_id, builddir):
+    for full_path in find_desktop_files(builddir):
+        cp = ConfigParser.RawConfigParser()
+        cp.read([full_path])
+        try:
+            icon = cp.get('Desktop Entry', 'Icon')
+        except ConfigParser.NoOptionError:
+            icon = None
+
+        # Does it have an icon?
+        if icon and not icon.startswith(app_id):
+            found_icon = False
+
+            # Rename any matching icons
+            for icon_file in find_icons(builddir, icon):
+                shutil.copy(icon_file,
+                            os.path.join(os.path.dirname(icon_file),
+                                         add_app_prefix(app_id, icon, os.path.basename(icon_file))))
+                found_icon = True
+
+            # If we renamed the icon, change the desktop file
+            if found_icon:
+                subprocess.check_call(['desktop-file-edit',
+                                       '--set-icon',
+                                       add_app_prefix(app_id, icon, icon), full_path])
+
+        # Is the desktop file not prefixed with the app id, then prefix it
+        basename = os.path.basename(full_path)
+        if not basename.startswith(app_id):
+            shutil.move(full_path,
+                        os.path.join(os.path.dirname(full_path),
+                                     add_app_prefix(app_id,
+                                                    basename[:-len('.desktop')],
+                                                    basename)))
+
+
+class FlatpakCreateOciPlugin(PostBuildPlugin):
```

Comment:
I'm not sure I entirely understand the distinction - the pre-publish plugins are run first, but the post-build plugins need a particular ordering from the build.json anyways. But the thought did occur to me "since it modifies the image, it's like squash"

The reason that I didn't do it that way is the post_rpmqa plugin. The "modified image" created by flatpak_create_oci no longer has rpm in it, so you can't run 'rpm -qa' in it - so from that perspective, post_rpmqa should conceptually run *before* flatpak_create_oci. In practical terms, it doesn't matter, since the original image is still there, and post_rpmqa is just going to run against that.

I'll go ahead and change it.